src/main/java/ch/njol/skript/sections/SecWhile.java

```diff
@@ -53,55 +66,143 @@
 @Since("2.0, 2.6 (do while)")
 public class SecWhile extends LoopSection {
 
-	static {
-		Skript.registerSection(SecWhile.class, "[:do] while <.+>");
+	private static final SkriptPattern DO_PATTERN = PatternCompiler.compile("do");
+
+	private enum WhileState{
+		NORMAL("[:do] while <.+>"),
+		ALL("[:do] while all"),
+		ANY("[:do] while (any|at least one of)"),
```

Comment:
i don't feel `at least one of` makes much sense here
just `at least one` would make more sense, or some other alternative

Reply:
Thats how it is on SecConditional, I can just make it optional